pkg/xray/validators.go

```diff
@@ -533,3 +533,7 @@ func minLength(length int) func(i interface{}, k string) ([]string, []error) {
 func inList(strings ...string) schema.SchemaValidateDiagFunc {
 	return validation.ToDiagFunc(validation.StringInSlice(strings, true))
 }
+
+var matchesHoursMinutesTime = validation.ToDiagFunc(
+	validation.StringMatch(regexp.MustCompile(`^([0-1]?[0-9]|2[0-3]):[0-5][0-9]$`), "Wrong format input, expected valid hour:minutes form"),
```

Comment:
Wait a minute - this can't be right

SO, `9:00` is ok? Is that AM or PM? 

You're not testing your time validation sufficiently. Please test:

- [ ] `09:00` -`09:59` pass
- [ ] `h:mm` fails
- [ ] `00:00` - `09:59` pass
- [ ] Anything greater than 23:59 fails


The rest of the regex looks ok, but you need some more edge case testing. For a format of `HH:mm` it has to be in 24h format. If needs be, you can break this down into a few regex for clarity and use `validation.all` to combine them. Also, your error message isn't specific enough. It needs: `HH:mm`

Reply:
`9:00` is ok by Xray. If set like this, it's returned as-is. It's 24 hrs format, but for some reason missing 0 is ok. The format description was added to the docs. 
The error message is what API returns, but I'll add the `HH:mm` format mention.
Added several test scenarios.

Reply:
I don't get it - it is enforced